A importância de um código de ética na construção de chatbots e inteligência artificial (#5)
ai-girls/dev-to-posts: posts/A-importancia-de-um-codigo-de-etica-na-construcao-de-chatbots-e-IA.md

# A importância de um código de ética na construção de chatbots e em IA

chatbot, ArtificialIntelligence, ethic 

Você já deve ter assistido o filme "Exterminador do futuro" ou  "I, Robot", ambos são muitos famosos e falam sobre o extermínio da raça humana através dos robôs, e isso acarretou em medo e receio das pessoas em confiarem na Inteligência Artificial. Portanto entusiastas, desenvolvedores e empresas assumem a importância de um código de ética na construção dessas inteligências. 

> Ao longo dos últimos anos, muitos investigadores, ativista expuseram os inúmeros erros, preconceitos e uso indevido da tecnologia, como por exemplo um modelo usado em casos criminais norte Americanos para prever quais criminosos têm maior probabilidade de reincidência e assim os juízes tomam as decisões baseados nele, chamado [COMPAS](https://medium.com/thoughts-and-reflections/racial-bias-and-gender-bias-examples-in-ai-systems-7211e4c166a1) que demonstrou carregar viés racial ao fazer essa classificação, onde a cor da pele da pessoa acabava sendo um fator para ser considerado criminoso de alto risco.

Reviewer: ```suggestion
Ao longo dos últimos anos, muitos investigadores, ativista expuseram os inúmeros erros, preconceitos e uso indevido da tecnologia, como por exemplo, um modelo usado em casos criminais Norte Americanos para prever quais criminosos têm maior probabilidade de reincidência e com isso os juízes tomam as decisões baseadas nele. Esse sistema foi chamado de [COMPAS](https://medium.com/thoughts-and-reflections/racial-bias-and-gender-bias-examples-in-ai-systems-7211e4c166a1) e demonstrou carregar viés racial ao fazer as suas classificações, onde a cor da pele da pessoa acabava sendo um fator para ser considerado criminoso de alto risco.
```

> # A importância de um código de ética na construção de chatbots e IA

Reviewer: ```suggestion
# A importância de um código de ética na construção de chatbots e em IA
```
O que tu acha?

> #chatbot, #ArtificialIntelligence, #ethic 

Reviewer: Aqui pode ser sem o `#`
```suggestion
chatbot, ArtificialIntelligence, ethic 
```

funciona assim no dev.to né? (sinceramente não sei, da ultima vez fiquei uma eternidade tentando colocar as tags).

Author: boa, é sem #

>“Podemos alterar, copiar, encerrar, apagar ou utilizar programas de computador tanto quanto nos agradar [...] 
As restrições morais a que estamos sujeitos em nossas relações com sistemas contemporâneos de IA são todas baseadas em nossas responsabilidades para com outros seres”
 (Nick Bostrom - “The Ethics of Artificial Intelligence”)

> <img src="https://github.com/lauraDamacenoAlmeida/dev-to-posts/blob/master/posts/Captura%20de%20Tela%202020-03-30%20a%CC%80s%2015.58.51.png?raw=true">

Reviewer: ```suggestion
<img src="https://github.com/lauraDamacenoAlmeida/dev-to-posts/blob/master/posts/Captura%20de%20Tela%202020-03-30%20a%CC%80s%2015.58.51.png?raw=true">
```
Um comentário sobre esse path... a imagem que aparece no Markdown está vindo desse caminho ai, no caso, ele está vindo do seu projeto. Se você apagar esse projeto ou mudar o nome do arquivo, a imagem vai deixar de aparecer pra gente.

O que eu recomendo fazer, que seria uma opção (se tiver outra pode dizer). Renomear a imagem que você subiu, com um nome mais curto e sem espaços (`imagem-blabla.png`) e colocar essa imagem em uma diretório `posts/anexos/`. Além disso, seria preciso renomear esse path, para passar o caminho que a imagem estará nesse projeto (quando o PR for aceito).

> Ao longo dos últimos anos, muitos investigadores, ativista expuseram os inúmeros erros, preconceitos e uso indevido da tecnologia, como por exemplo o reconhecimento facial utilizado pelas autoridades de Nova York que já penalizou inúmeras pessoas indevidamente, onde ela rotula as pessoas como criminosas ou não.

Reviewer: ```suggestion
Ao longo dos últimos anos, muitos ativista expuseram os inúmeros erros, preconceitos e uso indevido da tecnologia. Como por exemplo o reconhecimento facial utilizado pelas autoridades de Nova York que já penalizou inúmeras pessoas indevidamente, rotulando as pessoas como criminosas ou não.
```
teria links para colocar de referência? Seria legal para as pessoas verem a noticia (tem gente que acha que isso não acontece).

Author: Coloquei o link

Author: E eu reescrevi o parágrafo

> Dois pesquisadores da Universidade de Stanford treinaram um algoritmo de IA para adivinhar as orientações sexuais das pessoas com base em fotografias. Eles descobriram que era mais preciso do que os humanos que tentaram a tarefa. Os cientistas dizem que seu objetivo era provar que era possível criar tal algoritmo.

Reviewer: ```suggestion
Outro caso, curioso foi o de dois pesquisadores da Universidade de Stanford que treinaram um algoritmo de IA para adivinhar as orientações sexuais das pessoas com base em fotografias. Eles descobriram que era mais preciso do que os humanos que tentaram a tarefa. Os cientistas dizem que seu objetivo era provar que era possível criar tal algoritmo. Porém .... (adicionar o paragrafo abaixo)
```
Mesma coisa aqui, seria legal o link de referência.

> Se os consumidores acessarem alguma vez, alguns podem visar pessoas em suas comunidades e atormentá-las com insultos homofóbicos, nunca considerando a validade dos dados usados para criar a tecnologia.

Reviewer: ```suggestion
Se os usuários desse sistema acessarem alguma vez,  e forem mal intencionados, eles podem visar pessoas em suas comunidades e atormentá-las com insultos homofóbicos. Sendo assim, os autores do projeto não consideraram a validade dos dados usados para criar a tecnologia.
```
Essa ultima frase eu alterei, mas sinceramente, não sei direito o que ela estava querendo passar.

> Ou, se os policiais usassem algo semelhante para procurar criminosos, os algoritmos tendenciosos poderiam tornar as pessoas de certos grupos étnicos cada vez mais vulneráveis a crimes que eles não cometeram.

Reviewer: ```suggestion
O mesmo acontece no primeiro exemplo trazido, se os policiais usassem algo semelhante para procurar criminosos? Os algoritmos tendenciosos poderiam tornar as pessoas de certos grupos étnicos cada vez mais vulneráveis a crimes que eles não cometeram.
```

> Os esforços para se lidar com estes desafios crescentes concentram-se frequentemente na importância do binômio “Ética+ IA”.

Reviewer: ```suggestion
Os esforços para se lidar com estes desafios crescentes concentram-se frequentemente na importância do binômio **“Ética+ IA”**.
```

Só pra dar mais enfase.

>Então como nós desenvolvedoras e entusiastas poderíamos construir inteligências que atingem seus objetivos e impactam a sociedade de uma maneira saudável?

## Como as máquinas aprendem?

> Existem 3 tipos de aprendizado de máquina Supervisionado, não supervisionado e por reforço. Neste blog como estamos falando mais sobre chatbots vamos falar sobre o supervisionado e não supervisionado.

Reviewer: ```suggestion
Existem 3 tipos de aprendizado de máquina **supervisionado**, **não supervisionado** e **por reforço**. Neste post como estamos falando mais sobre chatbots vamos falar sobre o **supervisionado** e **não supervisionado**.
```

> Supervisionado: Nós desenvolvedores temos controle sobre o que o bot diz, criando respostas em vez de permitir que os usuários o ensinem, ou seja nós temos um poder maior sobre o que nosso bot vai aprender.

Reviewer: ```suggestion
* Supervisionado: Nós desenvolvedores temos controle sobre o que o bot diz, criando respostas em vez de permitir que os usuários o ensinem, ou seja, nós temos um poder maior sobre o que nosso bot vai "aprender".
```

Reviewer: Faltou por o `*` para criar uma lista. Ou você acha melhor não colocar como uma lista?

Author: acho melhor sem lista, coloquei eles como H4 agora

> •	Vantagens: Você sabe exatamente como ele vai responder e o bot não pode ser corrompido, a menos que você treine ele com dados corrompidos.

Reviewer: ```suggestion
     * Vantagens: Você sabe exatamente como ele vai responder e o bot não pode ser corrompido, a menos que você treine ele com dados corrompidos.
```

Mesma correção para os pontinhos pretos a baixo.

Reviewer: E faltou corrigir essa parte aqui tbm (e as outras partes que essa bolinha preta aparece).

•	Desvantagens: É mais demorado e criar um bot convincente leva muito tempo.

> Aprendizado não supervisionado: O bot é educado por seus usuários e não pelo desenvolvedor.

Reviewer: ```suggestion
* Aprendizado não supervisionado: O bot é educado por seus usuários e não pelo desenvolvedor.
```

•	Vantagens: Os usuários fazem o trabalho de treinar e ensinar o bot e você não precisa se preocupar em gastar tempo atualizando-o.

> •	Desvantagens: O seu bot vai desenvolver uma personalidade inconsistente e você pode não ter conhecimento do que está sendo ensinado. Na pior das hipóteses, ele se transforma em um software desagradável, racista, sexista e homofóbico.

Reviewer: Eu ia falar pra por o link do bot racista da microsoft kkkk mas ta la em baixo.

> Se você quiser saber mais a fundo sobre esses tipos de aprendizados, [clique aqui.]( https://dev.to/aigirlsbr/afinal-o-que-e-machine-learning-ih5)

Reviewer: HUHUHUHUHU!!!

## Tay e o aprendizado não supervisionado

> <img src="https://github.com/lauraDamacenoAlmeida/dev-to-posts/blob/master/posts/Captura%20de%20Tela%202020-03-30%20a%CC%80s%2015.58.36.png?raw=true">

Reviewer: Mesmo comentário sobre a imagem lá de cima.

Reviewer: Uma coisa que eu esqueci... essas imagens estão na internet não? Porque pode passar a url delas também.

Author: ajustei as imagens

> A Tay foi um chatbot criado pela Microsoft em março de 2016, onde ela interagia e aprendia com os tweets dos usuários na rede social. E sem o tratamento correto na base de aprendizado, ela acabou em menos de 24 horas se tornando homofóbica, racista e tudo mais.

Reviewer: Coloca alguns links de notícias. 
é bom ter referências.

> E isso fez com que muitos desenvolvedores e empresas questionassem o aprendizado 100% não supervisionado, e como alternativa o aprendizado semi-supervisionado, onde teríamos um tratamento prévio dos dados que serão aprendidos pelo bot, por exemplo no caso da Tay, identificar palavrões, palavras homofóbicas, racistas e retirassem elas da base de conhecimento.

Reviewer: ```suggestion
E isso fez com que muitos desenvolvedores e empresas questionassem o aprendizado 100% não supervisionado. Surgindo como alternativa o aprendizado semi-supervisionado, onde teríamos um tratamento prévio dos dados que serão aprendidos pelo bot. Por exemplo no caso da Tay seria introduzido uma etapa de, identificar palavrões, palavras homofóbicas, racistas, às retirando da base de conhecimento.
```

Reviewer: O que tu acha dessa sugestão?

## Mas como garantir a ética nos chatbots e inteligências artificiais?
<img src ="https://media.giphy.com/media/NoCbUpxL1qzCw/giphy.gif">

Eu fiz uma pesquisa sobre os principais códigos de ética que as grandes empresas de inteligência artificial fizeram, entre elas [Google](https://ai.google/principles/), [Microsoft](https://www.microsoft.com/en-us/research/uploads/prod/2018/11/Bot_Guidelines_Nov_2018.pdf) e [IBM](https://www.ibm.com/blogs/watson/2017/10/the-code-of-ethics-for-ai-and-chatbots-that-every-brand-should-follow/), e elenquei alguns princípios que elas têm em comum:

> •	Prevenção

Reviewer: Coloca o * no lugar dessas bolinhas.

•	Transparência

•	Confiabilidade

•	Privacidade

•	Igualdade

### Prevenção

> Muitas empresas não pensam em prevenção de abusos por parte dos usuários para o chatbot, por exemplo: reconhecer palavrões, ameaças de morte ou até racismo.

Reviewer: Um comentário também para os parágrafos a baixo... acho que dava pra juntar esses três mini parágrafos de uma linha, em um paragrafo só.

Depois do acontecimento com a Tay, a Microsoft logo tirou-a do ar e criou um novo chatbot a Zo, que tem o tratamento na base de conhecimento onde evita os termos da lista negra. 

> Algumas bibliotecas de lista negra de código aberto são mantidas e são usadas por agentes de conversação virtual para evitar determinados diálogos, um exemplo disso é o [Wordfilter]( http://tinysubversions.com/2013/09/new-npm-package-for-bot-makers-wordfilter/).

Reviewer: ```suggestion
Algumas bibliotecas de lista negra de código aberto são mantidas e usadas por agentes de conversação virtual para evitar determinados diálogos, um exemplo disso é o [Wordfilter]( http://tinysubversions.com/2013/09/new-npm-package-for-bot-makers-wordfilter/).
```

### Transparência

Seja transparente sobre o fato de usar bots como parte de seu produto ou serviço.

> É mais provável que os usuários confiem em uma empresa transparente e próxima sobre o uso de tecnologia, e um bot é mais provável de ser confiável se os usuários entenderem que o bot está trabalhando para servir suas necessidades e é claro sobre têm suas limitações.

Reviewer: ```suggestion
É mais provável que os usuários confiem em uma empresa transparente sobre o uso de tecnologia. Um bot é mais provável de ser confiável se os usuários entenderem que o bot está trabalhando para servir suas necessidades, além de, deixar claro sobre suas limitações.
```
O ajuste dessa ultima linha, não sei se está passando o que você queria passar com a ultima frase.

Reviewer: E esse comentário, tu concorda com essa alteração?

> Uma vez que os designers podem dotar seus bots com “personalidade” e capacidades de linguagem natural, é importante transmitir aos usuários que eles não estão interagindo com outra pessoa e sim com um bot. Existem várias opções de design e isso pode ser feito para que não prejudique a experiência do usuário.

Reviewer: Pode por links?

### Confiabilidade
O desempenho dos sistemas bot baseados em IA pode variar desde o desenvolvimento até a implementação e o tempo que o bot é lançado para novos usuários e em novos contextos, é importante continuamente monitorar a confiabilidade.
Seja transparente sobre a confiabilidade do bot, apresente resumos de desempenho do sistema ou de em um contexto específico.

> É importante sempre pedir feedback aos usuários referente as interações que ele teve, pois isso nos ajuda a entender melhor onde nosso bot está errando e ajusta-lo.

Reviewer: ```suggestion
O desempenho dos sistemas baseados em IA podem variar desde o desenvolvimento até a implementação, além, do tempo que o bot é lançado para novos usuários e em novos contextos, é importante continuamente monitorar a confiabilidade. 
Como fazer isso?! Seja transparente sobre a confiabilidade do bot, apresente resumos de desempenho do sistema ou de em um contexto específico. É importante sempre pedir feedbacks aos usuários referente as interações que ele teve, pois isso nos ajudará a entender melhor onde nosso bot está errando e ajustá-lo.
```

### Privacidade
Informe ao usuário que os dados serão coletados e como serão usados e obter consentimento do usuário e não colete mais dados pessoais do que o necessário.

> E aí vem um questionamento: “Temos uma IA está sendo utilizada para prevenir suicídios, mas até que ponto ela pode interferir nas decisões humanas?”

Reviewer: ```suggestion
Informe ao usuário que os dados serão coletados e como serão usados. Não esqueça de obter consentimento do usuário e não colete mais dados pessoais do que o necessário!
E aí vem um questionamento: “Temos uma IA que está sendo utilizada para prevenir suicídios, mas até que ponto ela pode interferir nas decisões humanas?”
```
Um questionamento massa, ele será debatido na palestra? Seria massa que ele não fosse apenas levantado aqui, mas que acontecesse em algum lugar :)

Author: Então eu ia comentar sobre ele, mas o texto ia ficar maior. A gente pode no futuro fazer um texto sobre isso, oq acha?

### Igualdade

> A possibilidade de sistemas baseados em IA perpetuarem preconceitos sociais existentes ou introduzirem novos vieses, é uma das principais preocupações identificadas pela comunidade de IA relacionada à rápida implantação da IA.

Reviewer: ```suggestion
A possibilidade de sistemas baseados em IA perpetuarem preconceitos sociais existentes ou introduzirem novos vieses, é uma das principais preocupações identificadas pela comunidade de IA relacionada à rápida implantação dela.
```

As equipes de desenvolvimento devem estar comprometidas em garantir que seus bots tratem todas as pessoas de forma justa.
Lute pela diversidade entre sua equipe de desenvolvimento. Empregando uma equipe diversificada focada no design, o desenvolvimento e o teste da tecnologia de bots ajudarão a garantir que seu bot funcione de forma justa.

> Preste atenção na base de dados que está sendo utilizada no treinamento da IA ou do chatbot para verificar se ela não está enviesada.

Reviewer: ```suggestion
As equipes de desenvolvimento devem estar comprometidas em garantir que seus bots tratem todas as pessoas de forma justa. Isso será alcançado com diversidade em sua equipe de desenvolvimento. Pois empregando uma equipe diversificada focada no design, no desenvolvimento e no teste da tecnologia, o bot terá mais chances de funcionar de forma justa.
Preste atenção na base de dados que está sendo utilizada no treinamento da IA ou do chatbot para verificar se ela não está enviesada.
```

>“As equipes de desenvolvimento devem estar comprometidas em garantir que seus bots tratem todas as pessoas de forma justa.”
( Microsoft, 2018, Bot Guidelines)

Até uma próxima,

AI Girl

Escritora: [Laura Damaceno de Almeida](https://www.linkedin.com/in/laura-damaceno/)

Siga a comunidade nas redes sociais!!

[LinkedIn](https://www.linkedin.com/company/ai-girls/)

[Facebook](https://www.facebook.com/aigirlsbr/)

[Instagram](https://www.instagram.com/aigirlsbrasil/)

> 

Reviewer: Oxe kkkk ficou um monte de linha em branco
